pkg/providers/ingress/translation/annotations/upstreamscheme/upstreamscheme.go

```diff
@@ -26,24 +27,24 @@ func NewParser() annotations.IngressAnnotationsParser {
 	return &upstreamscheme{}
 }
 
-// string in slice
-func inSchemes(a string) bool {
-	list := []string{"http", "https", "grpc", "grpcs"}
-
-	for _, b := range list {
-		if b == a {
-			return true
-		}
-	}
-	return false
+var schemeMap map[string]int = map[string]int{
+	"http":  1,
+	"https": 2,
+	"grpc":  3,
+	"grpcs": 4,
 }
 
 func (w *upstreamscheme) Parse(e annotations.Extractor) (interface{}, error) {
 	scheme := e.GetStringAnnotation(annotations.AnnotationsUpstreamScheme)
-
-	if inSchemes(scheme) {
+	_, ok := schemeMap[scheme]
```

Comment:
I thought we could ignore case?
both `HTTP` and `http` can be used.

You can convert the obtained configuration items to lowercase